components/constellation/constellation.rs

```diff
@@ -1483,9 +1476,26 @@ where
                 }
                 self.handle_panic(top_level_browsing_context_id, error, None);
             },
-            // Send frame tree to WebRender. Make it visible.
-            FromCompositorMsg::SelectBrowser(top_level_browsing_context_id) => {
-                self.send_frame_tree(top_level_browsing_context_id);
+            FromCompositorMsg::FocusBrowser(top_level_browsing_context_id) => {
+                if self.browsers.get(top_level_browsing_context_id).is_none() {
+                    return warn!(
+                        "{}: FocusBrowser on unknown top-level browsing context",
+                        top_level_browsing_context_id
+                    );
+                }
+                self.browsers.focus(top_level_browsing_context_id);
+                self.embedder_proxy.send((
+                    Some(top_level_browsing_context_id),
+                    EmbedderMsg::BrowserFocused(top_level_browsing_context_id),
+                ));
+                if !cfg!(feature = "multiview") {
+                    self.update_frame_tree_if_focused(top_level_browsing_context_id);
+                }
+            },
+            FromCompositorMsg::UnfocusBrowser => {
+                self.browsers.unfocus();
+                self.embedder_proxy
+                    .send((None, EmbedderMsg::BrowserUnfocused));
```

Comment:
Sorry, this is a serious nit, but how would you feel about "Blur" terminology to match the web?

Reply:
done in 4aea74fae2e6ae328104ce45fc63336941366cd7